compiler/rustc_hir/src/hir.rs

```diff
@@ -523,6 +523,40 @@ impl<'hir> GenericParam<'hir> {
             })
             .map(|sp| sp.shrink_to_hi())
     }
+
+    /// Returns the span of `:` after a generic parameter.
+    ///
+    /// For example:
+    ///
+    /// ```text
+    /// fn a<T:>()
+    ///       ^
+    ///       |      here
+    ///       here   |
+    ///              v
+    /// fn b<T       :>()
+    ///
+    /// fn c<T
+    ///
+    /// :>()
+    /// ^
+    /// |
+    /// here
+    /// ```
+    pub fn colon_span_for_suggestions(&self, source_map: &SourceMap) -> Option<Span> {
```

Comment:
This might possibly merge-conflict with #95970, since it's duplicated from that. Might want to wait on that landing and rebase.

Reply:
Done